src/Gaufrette/Adapter/AzureBlobStorage.php

```diff
@@ -403,4 +408,53 @@ private function guessContentType($content)
 
         return $fileInfo->buffer($content);
     }
+
+    /**
+     * Error message to be parsed.
+     *
+     * @param  ResponseInterface $response The response with a response body.
+     *
+     * @return string
+     */
+    protected static function parseErrorCode(ResponseInterface $response)
+    {
+        //try to parse using xml serializer, if failed, return the whole body
+        //as the error message.
+        try {
+            $sxml = new \SimpleXMLElement($response->getBody());
+            $data = static::_sxml2arr($sxml);
+
+            if (array_key_exists('Code', $data)) {
+                $errorCode = $data['Code'];
+            } else {
+                $errorCode = $response->getReasonPhrase();
```

Comment:
You don't need this `else` clause as soon as you define `$response->getResponsePhrase()` as a default value for `$errorCode`, at the top of this method. By doing this, you can also delete the assignment in the `catch`.